src/main/java/appeng/client/gui/implementations/GuiUpgradeable.java

```diff
@@ -150,12 +162,24 @@ public void drawBG(final int offsetX, final int offsetY, final int mouseX, final
         this.handleButtonVisibility();
 
         this.bindTexture(this.getBackground());
+
         this.drawTexturedModalRect(offsetX, offsetY, 0, 0, 211 - 34, this.ySize);
         if (this.drawUpgrades()) {
             this.drawTexturedModalRect(offsetX + 177, offsetY, 177, 0, 35, 14 + this.cvb.availableUpgrades() * 18);
         }
         if (this.hasToolbox()) {
-            this.drawTexturedModalRect(offsetX + 178, offsetY + this.ySize - 90, 178, this.ySize - 90, 68, 68);
+            switch (this.getToolboxSize()) {
+                case 3 -> this
+                        .drawTexturedModalRect(offsetX + 178, offsetY + this.ySize - 90, 178, this.ySize - 90, 68, 68);
+                case 5 -> {
+                    this.bindTexture(this.getAdvancedBackground());
+                    // It's too big, so move it up a little bit
+                    // Why Storage Bus will use this to draw all the slot?????
+                    this.drawTexturedModalRect(offsetX + 178, offsetY + this.ySize - 90 - 7, 0, 0, 104, 104);
```

Comment:
```suggestion
                    this.drawTexturedModalRect(offsetX + 178, offsetY + this.ySize - 90 - 7, 0, 0, 104, 104);
                    this.bindTexture(this.getBackground());
```
Each bus has his own gui texture that overrides in each gui implementation here you should switch back texture after overriding it to draw advanced toolbox

Reply:
thx, finally fixed it.